cmd/fpga_crihook/main.go

```diff
@@ -102,85 +114,123 @@ func (he *hookEnv) getFPGAParams(reader io.Reader) (string, string, error) {
 	}
 
 	afuImage, ok := content["afu-image"]
 	if !ok {
-		return fmt.Errorf("%s/%s: 'afu-image' field not found in the 'packager gbs-info' output", fpgaRegion, fpgaAfu)
+		return fmt.Errorf("%s/%s: 'afu-image' field not found in the 'packager gbs-info' output", params.region, params.afu)
 	}
 
 	interfaceUUID, ok := afuImage.(map[string]interface{})["interface-uuid"]
 	if !ok {
-		return fmt.Errorf("%s/%s: 'interface-uuid' field not found in the 'packager gbs-info' output", fpgaRegion, fpgaAfu)
+		return fmt.Errorf("%s/%s: 'interface-uuid' field not found in the 'packager gbs-info' output", params.region, params.afu)
 	}
 
 	acceleratorClusters, ok := afuImage.(map[string]interface{})["accelerator-clusters"]
 	if !ok {
-		return fmt.Errorf("%s/%s: 'accelerator-clusters' field not found in the 'packager gbs-info' output", fpgaRegion, fpgaAfu)
+		return fmt.Errorf("%s/%s: 'accelerator-clusters' field not found in the 'packager gbs-info' output", params.region, params.afu)
 	}
 
-	if canonize(interfaceUUID.(string)) != canonize(fpgaRegion) {
-		return fmt.Errorf("bitstream is not for this device: region(%s) and interface-uuid(%s) don't match", fpgaRegion, interfaceUUID)
+	if canonize(interfaceUUID.(string)) != params.region {
+		return fmt.Errorf("bitstream is not for this device: region(%s) and interface-uuid(%s) don't match", params.region, interfaceUUID)
 	}
 
 	acceleratorTypeUUID, ok := acceleratorClusters.([]interface{})[0].(map[string]interface{})["accelerator-type-uuid"]
 	if !ok {
-		return fmt.Errorf("%s/%s: 'accelerator-type-uuid' field not found in the 'packager gbs-info' output", fpgaRegion, fpgaAfu)
+		return fmt.Errorf("%s/%s: 'accelerator-type-uuid' field not found in the 'packager gbs-info' output", params.region, params.afu)
 	}
 
-	if canonize(acceleratorTypeUUID.(string)) != canonize(fpgaAfu) {
-		return fmt.Errorf("incorrect bitstream: AFU(%s) and accelerator-type-uuid(%s) don't match", fpgaAfu, acceleratorTypeUUID)
+	if canonize(acceleratorTypeUUID.(string)) != params.afu {
+		return fmt.Errorf("incorrect bitstream: AFU(%s) and accelerator-type-uuid(%s) don't match", params.afu, acceleratorTypeUUID)
 	}
 
 	return nil
 }
 
-func (he *hookEnv) programBitStream(fpgaRegion string, fpgaAfu string, fpgaBitStreamPath string) error {
+func (he *hookEnv) programBitStream(params *fpgaParams, fpgaBitStreamPath string) error {
 	output, err := he.execer.Command("fpgaconf", fpgaBitStreamPath).CombinedOutput()
 	if err != nil {
-		return fmt.Errorf("failed to program AFU %s to region %s: error: %v, output: %s", fpgaAfu, fpgaRegion, err, string(output))
+		return fmt.Errorf("failed to program AFU %s to region %s: error: %v, output: %s", params.afu, params.region, err, string(output))
 	}
 	return nil
 }
 
+func (he *hookEnv) getProgrammedAfu(deviceNum string) (string, error) {
+	afuIDPath := fmt.Sprintf(he.afuIDTemplate, deviceNum, deviceNum)
```

Comment:
I'd leave a message for future us that this code supports only one AFU per device to give us a clue when the hook starts misbehaving with more advanced hardware.

Reply:
do you mean adding a comment or print a message?

Reply:
a comment would suffice I think

Reply:
done